doc/user/formats.rst

```diff
@@ -130,3 +130,66 @@ fractional_ranges
         be masked if :math:`[\frac{i-0.5}{nchans}, \frac{i+0.5}{nchans}]`
         overlaps any of the ranges. Note that this means that channel 0
         is centred at 0.0 but extends below it.
+
+Primary beam
+------------
+The file contains the information in a compact Fourier-transform ("aperture
+plane") form. It comprises a 3-dimensional array J of Jones matrices. Another
+three arrays :math:`\nu`, :math:`y` and :math:`x` indicate the frequency (in
+Hz) and spatial position (in metres) of each sample along the axes.
+
+To determine the response at a sampled frequency :math:`\nu_k` and some
+direction, turn the direction into direction cosines :math:`l` and :math:`m`
+relative to the pointing centre. Then the Jones
+matrix for the response is
+
+.. math:: \frac{1}{|x|\cdot|y|}
+          \sum_{i,j} e^{-2\pi i (x_i l + y_j m)\nu_i/c} J_{k,j,i}.
+
+Note the reversed axis order in accessing :math:`J`.
+
+To sample at an intermediate frequency, use linear interpolation along the
+frequency axis in the aperture plane.
+
+See :class:`katsdpmodels.primary_beam.PrimaryBeam` and
+:class:`katsdpmodels.primary_beam.AltAzFrame` for a description of sign
+conventions. The Jones matrices correspond to :attr:`.OutputType.JONES_HV`.
+
+Attributes
+^^^^^^^^^^
+model_type
+    ``primary_beam``
+
+model_format
+    ``aperture_plane``
+
+antenna
+    The name of the antenna to which this model applies. Absent if this model
+    is not specific to a single antenna (the more generic ``model_target`` may
+    provide human-readable information about the range of applicable antennas).
+
+receiver
+    Serial number of the receiver to which this model applies. Absent if this
+    model is not specific to a single receiver.
+
+The `antenna` and `receiver` may be compared to the actual antenna
+and receiver identifiers in use to detect incorrect models (for example, if a
+receiver was swapped out but the model was not updated).
+
+Datasets
+^^^^^^^^
+x, y
+    1D arrays containing the aperture-plane coordinates of the samples. It is
+    highly recommended that the coordinates are evenly spaced.
+
+frequency
+    1D array of sampled frequencies.
+
+aperture_plane
+    5D array, with axes corresponding to (in order)
+
+    - row of Jones matrix (length 2)
+    - column of Jones matrix (length 2)
```

Comment:
Each of these data (aperture values) products are calculated independently of each other for each jones element, yet the sin/cos factors are shared.  Any shape or size of products should be allowed for computation simultaneously to allow for reuse of sin/cos terms. There may be many antennas needing to be computed and you do not necessarily want to loop over them when the task can be broadcast more efficiently using a GPU.

Reply:
I'm not sure what change you're suggesting. This PR is just for the file format. The computation API is in https://github.com/ska-sa/katsdpmodels/pull/24.

Reply:
In terms of file format the shape is [npol,nant,nchan,gridsize,gridsize], where pols are ['HH','HV','VH',VV'], and in terms of ants index -1 is array average. I will probably provide files that contain all antennas, plus array average. Not one file per antenna.

Reply:
I am assuming there will be a conversion step from files that make sense for holography to files that make sense for imaging.

At some point you'll need to tell me which way around "HV" and "VH" are. One of them is the response of the horizontal dipole to vertical polarisation, but I don't know which.

Reply:
In my opinion:
HH=Hfeed response due to H polarised signal
HV=Hfeed response due to V polarised signal
VH=Vfeed response due to H polarised signal
VV=Vfeed response due to V polarised signal